### PR TITLE
remove internal version api from kubectl

### PR DESCRIPTION
**Pull request by zjj2wry**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
/cc @liggitt  
**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/run_test.go [hunk 1]
```diff
@@ -232,8 +231,8 @@ func TestGenerateService(t *testing.T) {
 					Selector: map[string]string{
 						"run": "foo",
 					},
-					Type:            api.ServiceTypeClusterIP,
-					SessionAffinity: api.ServiceAffinityNone,
+					Type:            v1.ServiceTypeClusterIP,
+					SessionAffinity: v1.ServiceAffinityNone,
```
Comment (liggitt) [inline pkg/kubectl/cmd/run_test.go]: the unit test failure is legitimate... previously, this was going through defaulting. now, it's not, so Type:ClusterIP and SessionAffinity:None are no longer set

Comment (zjj2wry) [reply pkg/kubectl/cmd/run_test.go]: got it, updated

File: pkg/kubectl/cmd/run.go [hunk 2]
```diff
@@ -25,15 +25,15 @@ import (
...
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/watch"
-	api "k8s.io/kubernetes/pkg/apis/core"
-	coreclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/core/internalversion"
-	conditions "k8s.io/kubernetes/pkg/client/unversioned"
+	coreclient "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/kubernetes/pkg/client/conditions"
```
Comment (zjj2wry) [inline pkg/kubectl/cmd/run.go]: why this pkg not move into client-go, i feel it is helpful

**Comment by zjj2wry** [comment]
run denpendency attach and log, but they support internal version now, will create another PR fix them together.
```
AttachablePodForObject(object runtime.Object, timeout time.Duration) (*api.Pod, error)
LogsForObject(object, options runtime.Object, timeout time.Duration) (*restclient.Request, error)
```

**Comment by zjj2wry** [comment]
@liggitt thanks
ref issues #https://github.com/kubernetes/kubectl/issues/83

**Pull request merged**